client/core/core.go

```diff
@@ -128,6 +128,18 @@ func (dc *dexConnection) hasOrders(assetID uint32) bool {
 	return false
 }
 
+// hasActiveOrders checks whether there are any active orders for the dexConnection.
+func (dc *dexConnection) hasActiveOrders() bool {
+	dc.tradeMtx.RLock()
+	defer dc.tradeMtx.RUnlock()
+	for _, trade := range dc.trades {
+		if trade.metaData.Status < order.OrderStatus(order.MakerRedeemed) {
```

Comment:
A couple issue with this:
1. Casting from `MatchStatus` to `OrderStatus` isn't meaningful.
2. We can't do arithmetic reliably with `order.OrderStatus`.  We'd have to `switch` on `OrderStatus` to determine if it is active (active orders are `OrderStatusEpoch` and `OrderStatusBooked`).

Reply:
And an order can also be cancelled or executed, but still have active matches with `(*matchTracker).MetaData.Status < order.MakerRedeemed`. That needs to be checked as well.

Reply:
<img width="521" alt="Screen Shot 2020-05-28 at 02 22 21" src="https://user-images.githubusercontent.com/9800579/83063351-2e3e7e80-a08a-11ea-8fbe-b1c34aff1857.png">

There is three orders with diff status, but when I log the length of `dc.trades[order.OrderID].matches` always return 0